src/main/java/com/example/javaee_jpa_hibernate/repository/InvoiceRepository.java

```diff
@@ -0,0 +1,73 @@
+package com.example.javaee_jpa_hibernate.repository;
+
+import com.example.javaee_jpa_hibernate.model.Invoice;
+import com.example.javaee_jpa_hibernate.model.InvoiceBody;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static javax.transaction.Transactional.TxType.REQUIRED;
+import static javax.transaction.Transactional.TxType.SUPPORTS;
+
+@Transactional(SUPPORTS)
+public class InvoiceRepository implements CrudOperations {
+    @PersistenceContext(unitName = "Invoices")
+    private final EntityManager entityManager;
+
+    public InvoiceRepository(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public EntityManager getEntityManager() {
+        return entityManager;
+    }
+
+    @Override
+    @Transactional(REQUIRED)
+    public Invoice create(@NotNull Invoice invoice) {
+        entityManager.persist(invoice);
+        return invoice;
+    }
+
+    @Override
+    public Invoice findById(@NotNull @Min(1) @Max(Long.MAX_VALUE) Long id) {
```

Comment:
There is no need for validation of this parameter - it will make more sense in REST part